django_apscheduler/admin.py

```diff
@@ -86,45 +85,45 @@ def run_selected_jobs(self, request, queryset):
                 )
                 continue
 
-            scheduler.add_job(
+            if self._is_job_running(item.id):
+                msg_dict = {"job_id": item.id}
+                msg = _(
+                    "The job {job_id} is running now! Skipping execution..."
+                )
+                self.message_user(
+                    request, format_html(msg, **msg_dict), messages.WARNING
+                )
+                continue
+
+            self._scheduler.add_job(
                 django_job.func_ref,
                 trigger=None,  # Run immediately
                 args=django_job.args,
                 kwargs=django_job.kwargs,
                 id=django_job.id,
                 name=django_job.name,
-                misfire_grace_time=django_job.misfire_grace_time,
-                coalesce=django_job.coalesce,
+                # misfire_grace_time=django_job.misfire_grace_time,
+                misfire_grace_time=600,
```

Comment:
Why was the significance of this specific magic number of ‘600’?

Reply:
In my project, in case use selected multiple jobs to trigger at the same time and the jobs are too heavy to run in a single process, I just hard code the `max_workers=1` and roughly set the `misfire_grace_time=600` to walk around the 'EVENT_JOB_MISSED' issue.

It's not graceful enough to have the selected jobs run one by one.

Reply:
`max_workers` is 20 by default. Setting it to 1 means that you effectively lose all of the benefits of multi-threading or multi-processing (depending on the type of APScheduler executor that you are using). Not sure why that would be desirable unless you need your jobs to run in a specific sequence? There might be better ways of achieving that.

Increasing `misfire_grace_time` implies that your jobs will wait longer beyond their scheduled run time to kick off. Most users would prefer the default behavior: having `EVENT_JOB_MISSED` raised to let them know that there aren’t enough threads / processors available to keep up with the workload.